code/job.py

```diff
@@ -0,0 +1,69 @@
+from flask_restful import Resource, reqparse
+
+jobs = []
+
+class Job(Resource):
+    parser = reqparse.RequestParser()
+    parser.add_argument(
+        'company',
+        type=str,
+        required=True,
+        help="This field cannot be left blank"
+        )
+    parser.add_argument('salary')
+    parser.add_argument('status')
+
+    def post(self, title):
+
+        if self.check_for_job(title) is not None:
+            return{'message': "A job with name '{}' already exists".format(title)}, 400
+
+        data = Job.parser.parse_args()
+
+        job = {
+            'title': title, 
+            'company': data['company'],
+            'salary': data['salary'],
+            'status': data['status']
+            }
+        jobs.append(job)
+
+        return job, 201
+
+    def check_for_job(self, title):
+        for job in jobs:
```

Comment:
I try not to worry too much about optimising performance early on in development, however, this strikes me as potentially quite an egregious bit of code, as it will take longer and longer the more jobs you add to this list.

It's worth looking at using a dictionary, keyed on the job title. You can then use jobs.get(title) to retrieve the job. It will return None if one can't be found.

This actually has the added bonus of making this whole method one line, at which point, you could consider sacking off the method entirely, and just using the get method directly.

Reply:
Absolutely! I've changed jobs to a dictionary keyed on title. This method is no longer required as can now use using dict.get() instead.